bagitobjecttransfer/caais/forms.py

```diff
@@ -132,10 +132,13 @@ def save(self, commit=True):
         ''' Save the accession identifier input as an Identifier with the type
         name "Accession Identifier"
         '''
-        metadata = super().save(commit=commit)
+        metadata: Metadata = super().save(commit=commit)
 
         accession_id = self.cleaned_data['accession_identifier']
-        if not metadata.accession_identifier:
+
+        if accession_id and not metadata.accession_identifier:
```

Comment:
I think this block will be easier to understand if it's refactored like this:
```
if accession_id:
    if accession_id != metadata.accession_identifier:
        if not metadata.pk:
            metadata.save()  # Ensure the metadata instance is saved to get a primary key

        if not metadata.accession_identifier:
            # Create a new Identifier if it doesn't exist
            Identifier.objects.create(
                metadata=metadata,
                identifier_type='Accession Identifier',
                identifier_value=accession_id,
            )
        else:
            # Update the existing accession identifier
            metadata.update_accession_id(accession_id)
```

Reply:
How does this look? [d15c317](https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/227/commits/d15c3176ba4d211a8dfbc8d6de34e734968aae1f)

I generally prefer to avoid nested if statements, let me know if the new version is more readable.

Reply:
I agree, a triple nested if statement is not ideal. Yes, the new version is more readable.

I have a different concern - when the method `save` is called with `commit=False`, this part of the method:
```
if not metadata.pk:
            metadata.save()
```
commits to the save regardless of whether `commit` is `True` or `False`.

I made some suggested changes on some lines, could you check check whether the changes would reflect the desired behaviour?

Reply:
I accidentally deleted my comment for this, so I've written it here again.

I agree that a triple-nested if statement is not ideal. Yes, the new version is more readable.

I have a different concern, which I've written below.